packages/backend/src/database/migrations/20250807212731_add_custom_roles.ts

```diff
@@ -0,0 +1,184 @@
+import { Knex } from 'knex';
+import { type DbScopeInsert } from '../entities/roles';
+
+const ScopesTableName = 'scopes';
+const RolesTableName = 'roles';
+const ScopedRolesTableName = 'scoped_roles';
+
+export async function up(knex: Knex): Promise<void> {
+    // Create roles table
+    await knex.schema.createTable(RolesTableName, (table) => {
+        table
+            .uuid('role_uuid')
+            .primary()
+            .defaultTo(knex.raw('uuid_generate_v4()'));
+        table.string('name', 100).notNullable();
+        table.text('description').nullable();
+        table
+            .uuid('organization_uuid')
+            .notNullable()
+            .references('organization_uuid')
+            .inTable('organizations')
+            .onDelete('CASCADE');
+        table
+            .uuid('created_by')
+            .references('user_uuid')
+            .inTable('users')
+            .onDelete('SET NULL');
+        table.timestamp('created_at').defaultTo(knex.fn.now());
+        table.timestamp('updated_at').defaultTo(knex.fn.now());
+
+        table.unique(['name', 'organization_uuid']);
+    });
+
+    // Create scopes table
+    await knex.schema.createTable(ScopesTableName, (table) => {
+        table
+            .uuid('scope_uuid')
+            .primary()
+            .defaultTo(knex.raw('uuid_generate_v4()'));
+        table.string('resource', 100).notNullable();
+        table.string('action', 50).notNullable();
+        table.text('description').nullable();
+        table.boolean('is_commercial').notNullable();
+        table.specificType(
+            'name',
+            `text GENERATED ALWAYS AS (resource || ':' || action) STORED`,
+        );
+        table.timestamp('created_at').notNullable().defaultTo(knex.fn.now());
+
+        table.unique(['resource', 'action']);
+    });
+
+    // Create scoped_roles join table
+    await knex.schema.createTable(ScopedRolesTableName, (table) => {
+        table
+            .uuid('role_uuid')
+            .notNullable()
+            .references('role_uuid')
+            .inTable('roles')
+            .onDelete('CASCADE');
+        table
+            .uuid('scope_uuid')
+            .notNullable()
+            .references('scope_uuid')
+            .inTable('scopes')
+            .onDelete('CASCADE');
+        table.timestamp('granted_at').defaultTo(knex.fn.now());
+        table
+            .uuid('granted_by')
+            .notNullable()
+            .references('user_uuid')
+            .inTable('users');
+
+        table.index('role_uuid', 'idx_role_scopes_role');
```

Comment:
I don't think we need to add `idx_role_scopes_role`  , it will be created automatically